dash/development/_r_components_generation.py

```diff
@@ -502,14 +501,19 @@ def write_js_metadata(pkg_data, project_shortname, has_wildcards):
 
     os.makedirs("inst/deps")
 
-    for javascript in glob.glob("{}/*.js".format(project_shortname)):
-        shutil.copy(javascript, "inst/deps/")
+    for rel_dirname, _, filenames in os.walk(project_shortname):
+        for filename in filenames:
+            extension = os.path.splitext(filename)[1]
 
-    for css in glob.glob("{}/*.css".format(project_shortname)):
-        shutil.copy(css, "inst/deps/")
+            if extension not in [".css", ".js", ".map"]:
+                continue
 
-    for sourcemap in glob.glob("{}/*.map".format(project_shortname)):
-        shutil.copy(sourcemap, "inst/deps/")
+            target_dirname = os.path.join(os.path.join("inst/deps/", rel_dirname))
+
+            if not os.path.exists(target_dirname):
+                os.makedirs(target_dirname)
+
+            shutil.copy(os.path.join(rel_dirname, filename), target_dirname)
```

Comment:
So this is the real change - everything else is just for the nested/standard file structure test.

Instead of globing files w/o recursion, glob with recursion (using os.walk because glob.glob does not support the `recursive` named param in py27)